src/details/ArborX_DetailsAlgorithms.hpp

```diff
@@ -59,7 +59,14 @@ bool isValid(Point const &p)
 KOKKOS_INLINE_FUNCTION
 bool isValid(Box const &b)
 {
-  return isValid(b.minCorner()) && isValid(b.maxCorner());
+  using KokkosExt::isFinite;
+  for (int d = 0; d < 3; ++d)
+  {
+    auto const r_d = b.maxCorner()[d] - b.minCorner()[d];
+    if (r_d <= 0 || !isFinite(r_d))
```

Comment:
If both corners are infinities of the same size, their difference is going to be NaN. Comparison with NaN is always false, so the first clause would be false. I'm not sure what the result of `isFinite(NaN)`. Can you make sure and add a test that `isValid` is correct in this situation?

Reply:
Covered by https://github.com/arborx/ArborX/blob/603ec3970c2800cd6113b38c8ad8a0d9f4e28759/test/tstBoostGeometryAdapters.cpp#L160-L161

I can add a check in `test/tstDetailsAlgorithms.cpp` if you insist but `std::isfinite(NAN)` [returns `false`](https://en.cppreference.com/w/cpp/numeric/math/isfinite).